### PR TITLE
Update go-build-template scaffold to the latest.

### PR DESCRIPTION
**Pull request by MrHohn**
This is basically copy-pasting the latest version of https://github.com/thockin/go-build-template without any functional change. This helps introduce newly added features from upstream and get rid of a few stale items from the old scripts.

Just doing a refresh in the light of https://github.com/kubernetes-sigs/cluster-proportional-autoscaler/issues/131.

/assign @jingyuanliang

### CODE REVIEW

File: build/lint [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
```
Comment (jingyuanliang) [inline build/lint:1]: why do we have a separate non-suffixed lint here?

Comment (MrHohn) [reply build/lint]: Good catch! Removed now.

**Comment by MrHohn** [comment]
Thanks! Merging per the LGTM.

**Pull request merged**